app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -273,9 +273,15 @@ def raw_connect_v4(server, port, path, username, password, service, scheme = 'ht
       # Get the timeout from the configuration:
       timeout, = ems_timeouts(:ems_redhat, service)
 
-      # Create the connection:
+      url = URI::Generic.build(
+        :scheme => scheme,
+        :host   => server,
+        :port   => port,
+        :path   => path
+      )
+
       OvirtSDK4::Connection.new(
-        :url      => "#{scheme}://#{server}:#{port}#{path}",
+        :url      => url.to_s,
```

Comment:
@jhernand Yeah, I'm ok with strings.  I'm just wondering if making our interfaces accept URI objects would enable us to do IPv6 correctly by following convention.